tests/nvmeof/workflows/initiator.py

```diff
@@ -14,6 +14,32 @@ def __init__(self, node, gateway):
         self.gateway = gateway
         self.discovery_port = 8009
 
+    def fetch_nvme_devices_for_gw(self, gateway):
+        """Fetch nsids for all devices optimised for the given gateway
+
+        Returns:
+            nsids for which the given gateway is optimised.
+            empty list if the gateway is inaccessible for all devices
+        """
+        out, _ = self.node.exec_command(
+            cmd="nvme list --output-format json", sudo=True
```

Comment:
This is already part of nvme initiator lib, if not add this lib method.

Reply:
I saw the lib method now. Will make changes accordingly.

Reply:
This might not be required when we go with the uuid approach as discussed

Reply:
Changed the method altogether as discussed